runner/diff.go

```diff
@@ -63,3 +71,167 @@ func diffRunner(ctx context.Context, test *model.Test, sub model.KeyedSubmission
 	// TODO(flowlo): Save result.
 	return nil
 }
+
+func IODiffRun(ctx context.Context, in, out string, sub model.KeyedSubmission) (testResult model.DiffTestResult, err error) {
+	image := newImage(sub.Language)
+
+	if err = prepareImage(ctx, image); err != nil {
+		return
+	}
+
+	var v *docker.Volume
+	if v, err = createDockerVolume(sub.Code.Bucket + "/" + path.Dir(sub.Code.Name)); err != nil {
+		return
+	}
+
+	var c *docker.Container
+	c, err = dc.CreateContainer(docker.CreateContainerOptions{
+		Config: &docker.Config{
+			Image:     newImage(sub.Language),
+			OpenStdin: true,
+			StdinOnce: true,
+		},
+		HostConfig: &docker.HostConfig{
+			Privileged: false,
+			Memory:     0, // TODO(flowlo): Limit memory
+			Binds:      []string{v.Name + ":/run"},
+		},
+	})
+	if err != nil {
+		return
+	}
+
+	var stdin io.ReadCloser
+	stdin, err = storage.NewReader(ctx, "coduno-tests", in)
```

Comment:
use [this](https://github.com/coduno/api/blob/master/util/storage.go#L32) as context